### PR TITLE
Hb user events

### PR DESCRIPTION
**Pull request by HaroldBernard**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,3 +43,17 @@ Go into the package directory e.g. `user` then use `go test -v` within that dire
 
 Specify the user ID for the demo account to automatically share with a new signup with VCA.
 ```
+
+## Running the service locally
```
Comment (toddkazakov) [inline README.md:47]: This is not required anymore - it can be deleted.

File: user/api.go [hunk 2]
```diff
@@ -435,53 +431,58 @@ func (a *Api) GetUserInfo(res http.ResponseWriter, req *http.Request, vars map[s
...
 	}
 
-	var id string
-	if td.IsServer == true {
-		id = vars["userid"]
-		a.logger.Println("operating as server")
-	} else {
-		id = td.UserId
+	var requiresPassword bool
+	if !tokenData.IsServer {
```
Comment (toddkazakov) [inline user/api.go:443]: This commit shouldn't be in this PR.

File: user/mongoStoreClient.go [hunk 3]
```diff
@@ -102,6 +102,12 @@ func (msc *MongoStoreClient) EnsureIndexes() error {
 				SetExpireAfterSeconds(0).
 				SetBackground(true),
 		},
+		{
+			Keys: bson.D{{Key: "userId", Value: 1}},
```
Comment (toddkazakov) [inline user/mongoStoreClient.go:106]: This shouldn't be part of this PR.

File: README.md (hunk 1 above)
Comment (HaroldBernard) [reply README.md]: removed

File: user/mongoStoreClient.go (hunk 3 above)
Comment (HaroldBernard) [reply user/mongoStoreClient.go]: removed

File: user/api.go (hunk 2 above)
Comment (HaroldBernard) [reply user/api.go]: commented out here and in the tests

File: user/api.go [hunk 4]
```diff
@@ -232,6 +232,11 @@ func (a *Api) CreateUser(res http.ResponseWriter, req *http.Request) {
 			a.logMetricForUser(newUser.Id, "usercreated", sessionToken.ID, map[string]string{"server": "false"})
 			res.Header().Set(TP_SESSION_TOKEN, sessionToken.ID)
 			a.sendUserWithStatus(res, newUser, http.StatusCreated, false)
+			if err := a.userEventsNotifier.NotifyUserCreated(req.Context(), *newUser); err != nil {
```
Comment (toddkazakov) [inline user/api.go:235]: `a.sendUserWithStatus(res, newUser, http.StatusCreated, false)` must be placed after this block

File: user/api.go [hunk 5]
```diff
@@ -274,6 +279,11 @@ func (a *Api) CreateCustodialUser(res http.ResponseWriter, req *http.Request, va
 		} else {
 			a.logMetricForUser(newCustodialUser.Id, "custodialusercreated", sessionToken, map[string]string{"server": strconv.FormatBool(tokenData.IsServer)})
 			a.sendUserWithStatus(res, newCustodialUser, http.StatusCreated, tokenData.IsServer)
+			if err := a.userEventsNotifier.NotifyUserCreated(req.Context(), *newCustodialUser); err != nil {
```
Comment (toddkazakov) [inline user/api.go:282]: `a.sendUserWithStatus(res, newCustodialUser, http.StatusCreated, tokenData.IsServer)` must be placed after this block

File: user/api.go [hunk 6]
```diff
@@ -367,24 +377,13 @@ func (a *Api) UpdateUser(res http.ResponseWriter, req *http.Request, vars map[st
...
-					if err := a.userEventsNotifier.NotifyUserUpdated(req.Context(), *originalUser, *updatedUser); err != nil {
-						a.logger.Println(http.StatusInternalServerError, err.Error())
-						res.WriteHeader(http.StatusInternalServerError)
-						return
-					}
-				}
-			}
 			a.logMetricForUser(updatedUser.Id, "userupdated", sessionToken, map[string]string{"server": strconv.FormatBool(tokenData.IsServer)})
 			a.sendUser(res, updatedUser, tokenData.IsServer)
+			if err := a.userEventsNotifier.NotifyUserUpdated(req.Context(), *originalUser, *updatedUser); err != nil {
```
Comment (toddkazakov) [inline user/api.go:382]: `a.sendUser(res, updatedUser, tokenData.IsServer)` must be placed after this block

File: user/events.go [hunk 7]
```diff
@@ -89,22 +86,22 @@ func NewUserEventsHandler(store Storage) (events.UserEventsHandler, error) {
...
-		errs = append(errs, err)
-		log.Printf("Error deleteting user %v: %v", payload.UserID, err)
-		failedEvents.WithLabelValues(payload.GetEventType(), ShorelineUserEventHandlerName, RemoveUserOperationName)
-	}
-	if len(errs) == 1 {
-		return errs[0]
-	} else if len(errs) > 1 {
-		return errors.New(fmt.Sprintf("multiple errors occurred while deleting user: %v", errs))
-	}
+	// var errs []error
```
Comment (toddkazakov) [inline user/events.go:89]: You can just delete this for now

File: shoreline.go [hunk 8]
```diff
@@ -142,32 +117,41 @@ func main() {
...
-	userapi := user.InitApi(config.User, logger, clientStore, highwater, marketoManager)
+	// Start logging kafka connection debug info
+	sarama.Logger = logger
+
+	kafkaConfig := events.NewConfig()
+	if err := kafkaConfig.LoadFromEnv(); err != nil {
+		log.Fatalln(err)
+	}
+	//I believe this should be removed at some point
+	kafkaConfig.SaramaConfig.Net.TLS.Config.InsecureSkipVerify = true
```
Comment (toddkazakov) [inline shoreline.go:138]: This can be removed now - make sure you pull the latest version of `go-common`

File: user/api.go (hunk 5 above)
Comment (toddkazakov) [reply user/api.go:282]: I suggest removing this event completely if it's not used by marketo.

File: user/api.go (hunk 4 above)
Comment (toddkazakov) [reply user/api.go:235]: I suggest removing this event completely if it's not used by marketo.

Comment (HaroldBernard) [reply user/api.go:235]: ok wasnt sure if I should add the notify to all create and update events. removing

File: user/api.go (hunk 5 above)
Comment (HaroldBernard) [reply user/api.go:282]: removing

File: user/api.go (hunk 6 above)
Comment (HaroldBernard) [reply user/api.go:371]: done

File: user/events.go (hunk 7 above)
Comment (HaroldBernard) [reply user/events.go]: done

File: shoreline.go (hunk 8 above)
Comment (HaroldBernard) [reply shoreline.go]: done

**Comment by toddkazakov** [comment]
@HaroldBernard is this still relevant?
